### PR TITLE
build: clean up roachtest scripts

### PR DESCRIPTION
**Pull request by tbg**
See individual commits for details.

Release note: None

### CODE REVIEW

**Review by bobvawter — APPROVED** [review]
This looks good at the final commit.  What changes, if any, are you planning on making in the TeamCity UI?

> Reviewed 3 of 4 files at r4, 1 of 1 files at r6.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/47418#-:-M4noL6m87tCFpBsWGk6:b-h7gl6y)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @tbg)

___
*[build/teamcity-nightly-roachtest.sh, line 8 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/47418#-M4nmshp5cbbdrINmqgl:-M4nmshp5cbbdrINmqgm:b-zckmnj) ([raw file](https://github.com/cockroachdb/cockroach/blob/d7c6bb17fac9eb61f46d785796576e07569bb377/build/teamcity-nightly-roachtest.sh#L8)):*
> ```Shell
> 
> # Note that when this script is called, the cockroach binary to be tested
> # already exists in the current directory.
> ```

Please add an `if [[ ! -x ./cockroach ]]` as necessary to echo a message to that effect and exit the script.
___
*[build/teamcity-nightly-roachtest.sh, line 13 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/47418#-M4nnC4q8vEDWAe06doA:-M4nnC4q8vEDWAe06doB:bhjbet9) ([raw file](https://github.com/cockroachdb/cockroach/blob/d7c6bb17fac9eb61f46d785796576e07569bb377/build/teamcity-nightly-roachtest.sh#L13)):*
> ```Shell
> 
> if [[ ! -f ~/.ssh/id_rsa.pub ]]; then
>   ssh-keygen -q -N "" -f ~/.ssh/id_rsa
> ```

Maybe also set the key's comment to something like `-C "roachtest-nightly $(date)"` or similar to identify the key in case it ever winds up lingering somewhere?

**Review by tbg — COMMENTED** [review]
Basically replacing the script step in Roachtest Nightly GCE and the AWS version with something like this

```
# NB: please please please make no changes in this script unless you
# also make them for the other clouds.

script_branch="${BRANCH_OVERRIDE-master}"
echo "Using build script from ${script_branch}"
git checkout "origin/${script_branch}"
git pull --ff-only origin "${script_branch}"
build_tag="$(git describe --abbrev=0 --tags --match=v[0-9]*)"
CLOUD=gce BUILD_TAG="${build_tag}" ./build/teamcity-nightly-roachtest.sh
```

I have this at the top of the existing script (behind an env var) for testing purposes now (GCE only). Folks in #engineering reminded me of the upcoming release and a desire to avoid breaking random things. When I'm convinced (via a manual run) that it all works I will flip the switch.

Thanks for the swift review.

bors r=bobvawter

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/47418#-:-M4npMO0BZa_wB8IM3Z2:bzewsm9)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @bobvawter)

___
*[build/teamcity-nightly-roachtest.sh, line 8 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/47418#-M4nmshp5cbbdrINmqgl:-M4nq0KlBvvofb7CqIbk:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/d7c6bb17fac9eb61f46d785796576e07569bb377/build/teamcity-nightly-roachtest.sh#L8)):*
<details><summary><i>Previously, bobvawter (Bob Vawter) wrote…</i></summary><blockquote>

Please add an `if [[ ! -x ./cockroach ]]` as necessary to echo a message to that effect and exit the script.
</blockquote></details>

Done.
___
*[build/teamcity-nightly-roachtest.sh, line 13 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/47418#-M4nnC4q8vEDWAe06doA:-M4nq0fjAT3qGf4dYybg:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/d7c6bb17fac9eb61f46d785796576e07569bb377/build/teamcity-nightly-roachtest.sh#L13)):*
<details><summary><i>Previously, bobvawter (Bob Vawter) wrote…</i></summary><blockquote>

Maybe also set the key's comment to something like `-C "roachtest-nightly $(date)"` or similar to identify the key in case it ever winds up lingering somewhere?
</blockquote></details>

Done.

**Comment by tbg** [comment]
Flake in acceptance/gossip/restart

bors r=bobvawter

File: build/teamcity-nightly-roachtest.sh [hunk 1]
```diff
@@ -4,6 +4,94 @@
 # apologies, you're going to have to dig around for them below or even better
 # yet, look at the job).
 
+# Note that when this script is called, the cockroach binary to be tested
+# already exists in the current directory.
+COCKROACH_BINARY="$PWD/cockroach.linux-2.6.32-gnu-amd64"
+if [[ ! -x ./cockroach ]]; then
```
Comment (jlinder) [inline build/teamcity-nightly-roachtest.sh:10]: Why check for `./cockroach` and not `COCKROACH_BINARY`? Particularly when only `COCKROACH_BINARY` is used elsewhere but `./cockroach` doesn't seem to be used.

Comment (tbg) [reply build/teamcity-nightly-roachtest.sh:10]: Thanks for catching this, this is just wrong. I copied from Bob's comment.
The script isn't used yet, I will fix this and submit a PR with all of the other bugs that no doubt it contains when I've given this a test run.

**Pull request merged**